src/esp/gfx/Renderer.cpp

```diff
@@ -23,109 +23,29 @@ namespace esp {
 namespace gfx {
 
 struct Renderer::Impl {
-  Impl(int width, int height)
-      : framebufferSize_(width, height),
-        colorBuffer_(),
-        depthBuffer_(),
-        objectIdBuffer_(),
-        depthRenderbuffer_(),
-        framebuffer_({{}, framebufferSize_}) {
+  Impl() {
     GL::Renderer::enable(GL::Renderer::Feature::DepthTest);
     GL::Renderer::enable(GL::Renderer::Feature::FaceCulling);
-    setSize(width, height);
   }
   ~Impl() { LOG(INFO) << "Deconstructing Renderer"; }
 
-  void setSize(int width, int height) {
-    framebufferSize_[0] = width;
-    framebufferSize_[1] = height;
-    colorBuffer_.setStorage(GL::RenderbufferFormat::SRGB8Alpha8,
-                            framebufferSize_);
-    depthBuffer_.setStorage(GL::RenderbufferFormat::R32F, framebufferSize_);
-    objectIdBuffer_.setStorage(GL::RenderbufferFormat::R32UI, framebufferSize_);
-    depthRenderbuffer_.setStorage(GL::RenderbufferFormat::Depth24Stencil8,
-                                  framebufferSize_);
-    framebuffer_ = GL::Framebuffer{{{}, framebufferSize_}};
-    framebuffer_
-        .attachRenderbuffer(GL::Framebuffer::ColorAttachment{0}, colorBuffer_)
-        .attachRenderbuffer(GL::Framebuffer::ColorAttachment{1}, depthBuffer_)
-        .attachRenderbuffer(GL::Framebuffer::ColorAttachment{2},
-                            objectIdBuffer_)
-        .attachRenderbuffer(GL::Framebuffer::BufferAttachment::Depth,
-                            depthRenderbuffer_)
-        .mapForDraw({{0, GL::Framebuffer::ColorAttachment{0}},
-                     {1, GL::Framebuffer::ColorAttachment{1}},
-                     {2, GL::Framebuffer::ColorAttachment{2}}});
-    CORRADE_INTERNAL_ASSERT(
-        framebuffer_.checkStatus(GL::FramebufferTarget::Draw) ==
-        GL::Framebuffer::Status::Complete);
-  }
-
-  inline void renderEnter() {
-    framebuffer_.clear(GL::FramebufferClear::Color |
-                       GL::FramebufferClear::Depth);
-    framebuffer_.clearColor(1, Vector4{});
-    framebuffer_.clearColor(2, Vector4ui{});
-    framebuffer_.bind();
-  }
-
-  inline void renderExit() {}
-
   void draw(RenderCamera& camera, MagnumDrawableGroup& drawables) {
-    renderEnter();
-    camera.getMagnumCamera().setViewport(framebufferSize_);
     camera.draw(drawables);
-    renderExit();
   }
 
   void draw(sensor::Sensor& visualSensor, scene::SceneGraph& sceneGraph) {
     ASSERT(visualSensor.isVisualSensor());
 
     // set the modelview matrix, projection matrix of the render camera;
     sceneGraph.setDefaultRenderCamera(visualSensor);
+    sceneGraph.getDefaultRenderCamera().getMagnumCamera().setViewport(
+        visualSensor.renderingTarget()->framebufferSize());
 
     draw(sceneGraph.getDefaultRenderCamera(), sceneGraph.getDrawables());
```

Comment:
Maybe a bit out of topic, a couple questions to @mosra related to "instancing" :
Can a drawable be shared by a couple of scene nodes? (Background, e.g., there are 20 identical chairs in the scene, and I would like to load the mesh just once, create just 1 drawable, and make it shared among 20 scene nodes.)
Is there a way to quickly clone a sub-SceneGraph (along with all the attached drawables)? (e.g., a sub-SceneGraph representing a humanoid robot)

Reply:
> Can a drawable be shared by a couple of scene nodes?

No. But a couple of drawables can reference the same mesh/shader/texture multiple times, avoiding the need to load them more than once.

> Is there a way to quickly clone a sub-SceneGraph (along with all the attached drawables)? 

We discussed this with @hzyjerry a couple days ago on Slack, see there for details. Current plan is that he'll make it as part of #91, I think?